### PR TITLE
feat: refactor the code, improve usability and code structure

### PR DESCRIPTION
**Pull request by JackHamer09**
# What :computer:
* The entire app has been refactored to use [commander.js](https://github.com/tj/commander.js) as its command manager.
* By passing command options (flags), you can now execute each command instantly without encountering any prompts.
* Improved file structure. For example, all commands have been moved to the `commands/` folder. Various utility scripts have been added to the `utils/` folder.
* Renamed `confirm-withdraw` to `withdraw-finalize` and `create` to `create-project`.
* Enhanced command and option descriptions. Updated the README file.

# Why :hand:
* To make the app more scalable, a more efficient way to manage commands was required. [commander.js](https://github.com/tj/commander.js) was selected due to its popularity and ease of use.
* Introducing the possibility to pass command options makes the app easier to use with predefined parameters, allowing users to bypass prompts.
* To enhance code maintainability, the file structure was reorganized for greater convenience.
* The command `confirm-withdraw` was renamed to ensure consistency in language across all textual resources. `create` was renamed to be more descriptive and to reduce ambiguity.
* Both command and option descriptions were refined for clarity and to maintain a consistent language. The README file was revised to align with the latest changes.

# Evidence :camera:
image

# Notes :memo:
* Separate PR to docs will be created with updated zksync-cli description before merging this PR.
* Try it our with `npx zksync-cli@0.4.0-beta.1`

### CODE REVIEW

**Comment by dutterbutter** [comment]
This is awesome @JackHamer09 💯  

One issue I ran into while initially testing is when I run `npx zksync-cli@0.4.0-beta.1 create-project test-project` and select `Hardhat + Solidity` it creates the repo using the vyper template. Can you confirm that issue?

File: src/commands/create-project.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { track } from "../utils/analytics";
+import { optionNameToParam } from "../utils/helpers";
+import Logger from "../utils/logger";
+import zeek from "../utils/zeek";
+
+const templates = [
+  {
+    name: "Hardhat + Solidity",
+    value: "hardhat_solidity",
+    git: "https://github.com/matter-labs/zksync-hardhat-vyper-template",
```
Comment (dutterbutter) [inline src/commands/create-project.ts:17]: should be solidity template

Comment (JackHamer09) [reply src/commands/create-project.ts:17]: Thanks, my bad

File: src/commands/deposit.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import { isDecimalAmount, isAddress, isPrivateKey } from "../utils/validators";
+import zeek from "../utils/zeek";
+
+const chainOption = new Option("-c, --chain <chain>", "Chain to use").choices(
+  l2Chains.filter((e) => e.l1Chain).map((chain) => chain.network)
+);
+const amountOption = new Option("--amount <amount>", "Amount of ETH to deposit (eg. 0.1)");
+const recipientOption = new Option("--recipient <address>", "Recipient address on L2 (0x address)");
+
+type DepositOptions = {
```
[the remark this answers is not in the document]
Comment (JackHamer09) [reply src/commands/deposit.ts:27]: Makes sense, will work on it

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**